### PR TITLE
refactor(scripts): split `icons.ts` & add new commands

### PR DESCRIPTION
**Pull request by uncenter**
I found that having to use CLI flags was frustrating and it makes more sense to me at least to split each function into a separate file and run it as a separate script. I have done just that in this PR, and also recreated the `icons --all` behavior with `pnpm run icons`, which uses pnpm's [support for multiple scripts / regex](https://pnpm.io/cli/run#running-multiple-scripts) to run all the `icons:` subscripts.

### CODE REVIEW

**Comment by uncenter** [comment]
The diff is awful, as usual. I haven't changed a single line of code besides copying the original script into each file, removing the unrelated functions and function declaration wrapper, and removing unused imports.

File: package.json [hunk 1]
```diff
@@ -119,7 +119,10 @@
   "scripts": {
     "build": "tsx scripts/build.ts",
     "catwalk": "tsx scripts/catwalk.ts",
-    "icons": "tsx scripts/icons.ts",
+    "icons:generate": "tsx scripts/icons/generate.ts",
+    "icons:optimize": "tsx scripts/icons/optimize.ts",
+    "icons:preview": "tsx scripts/icons/preview.ts",
+    "icons": "pnpm run '/^icons:.*/'",
```
Comment (sgoudham) [inline package.json:125]: Do we know the order of the scripts that are ran? Is it the order in which they're defined in the scripts block,  alphabetical order, etc?

Comment (uncenter) [reply package.json:125]: Good point! I'll look into that ahah.

Comment (sgoudham) [reply package.json:125]: Discussed on discord, it's the order defined in the script block

**Pull request merged**